.github/workflows/clang-format.yaml

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+# SPDX-FileCopyrightText: Copyright 2022 Micron Technology, Inc.
+
+name: ClangFormat
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - master
+      - "v[0-9]+.[0-9]+"
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.h.in"
+      - "!subprojects/*"
+      - .github/workflows/clang-format.yaml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/hse-project/ci-images/fedora-37:${{ github.base_ref }}
+
+    steps:
+      - name: Checkout HSE
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Get clang-format
+        run: |
+          dnf install clang-tools-extra
```

Comment:
can you ensure version > 15 ?

Reply:
or maybe we can do that in the `clang-format.sh` script.

Reply:
you already check that in the script, but right now you have it set to 13

Reply:
Ahh... Right.  Go ahead and change it to 15 since 13 will produce slightly different output.